docs/explanation/lifecycle-layer.rst

```diff
@@ -0,0 +1,198 @@
+   drwxrwxr-x   2 root root  4096 ago 27  2022 cdrom
+   drwxr-xr-x  20 root root  5900 dez  7 19:57 dev
+   drwxr-xr-x 148 root root 12288 dez  7 15:15 etc
+   drwxr-xr-x   3 root root  4096 ago 27  2022 home
+   lrwxrwxrwx   1 root root     7 ago 27  2022 lib -> usr/lib
+   lrwxrwxrwx   1 root root     9 ago 27  2022 lib32 -> usr/lib32
+   lrwxrwxrwx   1 root root     9 ago 27  2022 lib64 -> usr/lib64
+   lrwxrwxrwx   1 root root    10 ago 27  2022 libx32 -> usr/libx32
+
+``bin`` is actually a symbolic link to ``usr/bin``. This is the usrmerge_, and
+it's been present in Ubuntu for many years now. Note that many other entries
+are also symlinks, like ``lib`` (to ``usr/lib``) and ``lib64``
+(to ``usr/lib64``).
+
+These two filesystems interact in a surprising way when stacked as OCI layers.
+If ``bin/pebble`` is added to the layer's archive as ``bin/pebble`` plus an
+entry for the ``bin/`` directory (which is a regular directory in the prime
+contents), once the two layers are stacked together in a container the ``bin/``
+directory from the "prime layer" will *overwrite* the ``bin -> usr/bin``
+symlink from the "base layer", which will make everything that assumed that
+the base binaries from ``usr/bin/`` would always be accessible through ``bin/``
+break.
+
+This issue is made much worse if the instead of breaking ``bin/`` we break the
+``lib*/`` symlinks. Consider:
+
+.. code-block:: bash
+
+   $ ldd /bin/bash
+   linux-vdso.so.1 (0x00007ffdf2af4000)
+   libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007f6053cbd000)
+   libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6053a00000)
+   /lib64/ld-linux-x86-64.so.2 (0x00007f6053e6b000)
+
+The ``bash`` binary links to multiple dynamic libraries, but has a hardcoded
+path to the ``/lib64/ld-linux-x86-64.so.2`` dynamic loader. This loader is the
+program that does the actual finding of dynamic dependencies at runtime, and in
+an Ubuntu system its actual location is at ``/usr/lib64/ld-linux-x86-64.so.2``.
+So if the ``/lib64 -> usr/lib64`` symlink is broken because the prime directory
+contains ``lib64`` as a regular directory, then the vast majority of the
+binaries in the final rock's base system will simply fail to run because their
+loader is no longer available at ``/lib64/ld-linux-x86-64.so.2``.
+
+To fix this, Rockcraft will take the base system into account when creating the
+archive for the prime layer. For instance, when considering ``bin/pebble``,
+Rockcraft will:
+
+#. Skip adding ``bin/`` as a regular directory, to avoid breaking the base
+   system, and
+#. Add ``bin/pebble`` as ``usr/bin/pebble`` in the layer archive.
+
+This can be seen in the logs:
+
+.. code-block:: text
+
+   (...)
+   Creating new layer
+   (...)
+   Skipping /root/prime/bin because it exists as a symlink on the lower layer
+   (...)
+   Adding to layer: /root/prime/bin/pebble as 'usr/bin/pebble'
+   (...)
+
+One final note is that none of this applies for rocks with ``bare`` bases, as
```

Comment:
well...what if I install the pebble snap? will rockcraft raise a conflict? or will the pruning system act?

Reply:
I would put this at the top, and specify this only applies to ubuntu bases